pkg/scale/varying_data_type.go

```diff
@@ -25,6 +25,7 @@ func (vdts *VaryingDataTypeSlice) Add(values ...VaryingDataTypeValue) (err error
 		copied := vdts.VaryingDataType
 		err = copied.Set(val)
 		if err != nil {
+			err = fmt.Errorf("%w: %s", ErrUnableToAddVaryingDataTypeValue, err)
```

Comment:
On which error to wrap...
This is a topic I myself struggled with for months (and might still be shared on this).
With the current Go standard library, we cannot 'wrap' multiple errors (in this case both ErrUnableToAddVaryingDataTypeValue and err), and so `errors.Is(err, errSentinel)` will only work with a single `errSentinel`. This imo sucks and doesn't make much sense, especially since they call it an error *chain*, but that's how it is (also feel free to 👍 https://github.com/golang/go/issues/53435).
Anyway, back to the solution, just wrap the bottom error pretty much almost always, unless there is a specific use case for example where we would have 2 `copied.Set` and we would need to know from which the error is coming from programmatically (with `errors.Is`).
In this case, you can thus change it to

```suggestion
			err = fmt.Errorf("adding VaryingDataTypeValue: %w", err)
```